### DISCUSSION TITLE
How to generate the right directory through tsx <h1> <h2> <h3>...

### DISCUSSION THREAD

**Discussion by pdsuwwz** [facebook/docusaurus#9628] [Q&A]
I want to know whether these titles `<h1>`  `<h2>`  `<h3>` written in `.tsx` can be automatically generated into the right toc 🙏

**Comment by slorber** [comment]
Hey

Not possible unfortunately: the TOC is extracted at build time. 

Anything in jsx/tsx requires rendering React to extract the headings, so it's hard to bring support for this.

**Reply by pdsuwwz** [reply]
Hi ~

Thank you for your prompt reply.

This is the first time for me to use docusaurus, and I may not understand some parts of it.

I'm actually not sure if there are people who have the same needs as me, and if there are, I still hope that such an API may appear in the future.

Such as the future may have a similar ` @ docusaurus/HeadingTitleXxxx ` this official component, so should be can through the component in the construction phase extraction, or some way better... Finally, Thank you for your help!
